src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionService.cs

```diff
@@ -0,0 +1,525 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Completion.Log;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal sealed class SerializableImportCompletionItem
+    {
+        public readonly string SymbolKeyData;
+        public readonly int Arity;
+        public readonly string Name;
+        public readonly Glyph Glyph;
+        public readonly string ContainingNamespace;
+
+        public SerializableImportCompletionItem(string symbolKeyData, string name, int arity, Glyph glyph, string containingNamespace)
+        {
+            SymbolKeyData = symbolKeyData;
+            Arity = arity;
+            Name = name;
+            Glyph = glyph;
+            ContainingNamespace = containingNamespace;
+        }
+    }
+
+    /// <summary>
+    /// Provides completion items for extension methods from unimported namespace.
+    /// </summary>
+    /// <remarks>It runs out-of-proc if it's enabled</remarks>
+    internal static partial class ExtensionMethodImportCompletionService
```

Comment:
calling this a completion service, but not actually deriving from completion-service is wonky.  this just confuses me.

Reply:
How about `ExtensionMethodImportCompletionHelper`?